frappe/__init__.py

```diff
@@ -1018,19 +1018,15 @@ def get_precision(
 	return get_field_precision(get_meta(doctype).get_field(fieldname), doc, currency)
 
 
-def generate_hash(txt: str | None = None, length: int | None = None) -> str:
-	"""Generates random hash for given text + current timestamp + random string."""
-	import hashlib
-	import time
-
-	from .utils import random_string
-
-	digest = hashlib.sha224(
-		((txt or "") + repr(time.time()) + repr(random_string(8))).encode()
-	).hexdigest()
-	if length:
-		digest = digest[:length]
-	return digest
+def generate_hash(txt: str | None = None, length: int = 56) -> str:
+	"""Generate random hash using best available randomness source."""
+	import math
+	import secrets
+
+	if not length:
+		length = 56
```

Comment:
Unless someone is manually sending length as a null-ish value, I don't think you need this check

Reply:
Yeah, just kept for compatibility cause previously we accepted `None`